internal/security/scrypt.go

```diff
@@ -0,0 +1,24 @@
+package security
+
+import (
+	"golang.org/x/crypto/scrypt"
+)
+
+// scrypt parameters.
+const (
+	costFactor            = 17
```

Comment:
So 17 is definitely too much. 

16 is still:
```
real    0m0.233s
user    0m0.154s
sys     0m0.081s
```

But I think that's more tolerable. With 17 - with multiple runs - it was between 0.5s and 0.8s. Sometimes close to one second is a lot.

With 16, it's between 0.1 and 0.4s. Still painful when it hits the upper part of the range, but more acceptable.

Yes, I use it lot, some AWS accounts have 20 mins timeout on session and I have to refresh my session with TOTP (I know, annoying)

Reply:
A potential resolution can be a "one time burden" on the user and when they create a new vault, we ask for a password and we can ask for `costFactor` and use that.

That way the user has control over the value and they can sacrifice time or security.

To set the value, it should be enough to change the code in 2 places:
1. `initStorage` in `storage.go`.
2. The `ChangePassword` command (that can be the way to change the factor, use the same password but set a new factor).

The factor can be stored at the beginning of the file in one byte.

Reply:
> Yes, I use it lot, some AWS accounts have 20 mins timeout on session and I have to refresh my session with TOTP (I know, annoying)

I was not expecting anyone to be running it that frequently. That definitely changes some assumptions I had when picking the value.

Reply:
> A potential resolution can be a "one time burden" on the user and when they create a new vault, we ask for a password and we can ask for costFactor and use that.

I dislike the idea of asking a random user to chose a cryptographic parameter.

> The factor can be stored at the beginning of the file in one byte.

I do agree with this, by storing it in the file we can avoid a future change to the file format. In fact, I should have just completely handed off the file encryption to age to begin with. I have a new work in progress for that https://github.com/nuxi/totp-cli/tree/age-based

Since I used the symmetric option in age, I still need to pick a cost factor for it though. I did some more digging into what an appropriate value is. Here are my notes:
* The original 2009 scrypt paper uses 14
* In 2017 the author said the original choice was based on a ~100ms runtime
* go docs (last updated 2017) suggest 15 by default (I got this wrong in the initla PR comment, I mixed it up with the OWASP guidelines)
* an expired IETF draft from 2020 also suggested 15
* OWASP said 16 in March 2021 and 17 as of January 2023
* age uses 18 by default, but its configurable

I don't really see a problem with using 15 given your use case of needing it frequently. I doubt this is really the weakest link in the program anymore and it can easily be changed in the future.

Does that sound good to you?

Reply:
15 sounds good to me.

> I dislike the idea of asking a random user to chose a cryptographic parameter.

yes that's ugly, I don't like it either :laughing: 
I would say go and store in the file as one byte, use 15 (or 16) by default and later we can add an extra command to change the value (like `set-encryption-cost-factor` intentionally long name so the user really pays attention to what it does).

The rest looks good and work as expected.